### PR TITLE
Feature/reworked Edit with angularjs

### PR DESCRIPTION
**Pull request by OKrain**
[Link to JIRA ticket](https://ssu-jira.softserveinc.com/browse/LR-181)

### Author.

- [x] I have added link to this PR to the JIRA
- [x] I have logged time in JIRA.

### Reviewers.

- [x] @anatoliiibrahimov 
- [x] @superNazo 
- [x] @vovka

### Issue summary.

*Rework edit with AngularJs*

### Fix summary. 

_Reworked edit with AngularJs_

### Testing approach.

There is unit test for editQuiz controller.

### CODE REVIEW

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 1]
```diff
@@ -32,3 +32,28 @@ quizzesControllers.controller("indexQuizCtrl", ["$scope", "Quizzes",
     };
   }
 ]);
+
+quizzesControllers.controller('editQuizCtrl',
```
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 2]
```diff
@@ -32,3 +32,28 @@ quizzesControllers.controller("indexQuizCtrl", ["$scope", "Quizzes",
...
+  ['$scope', '$routeParams', '$location', 'Quiz',
```
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 3]
```diff
@@ -32,3 +32,28 @@ quizzesControllers.controller("indexQuizCtrl", ["$scope", "Quizzes",
...
+        content: ''
+      });
+    };
+
+    $scope.editQuiz = function() {
+      Quiz.update({
+        id: $routeParams.quizId,
+        quiz: {
+          name: $scope.quiz.name,
+          questions_attributes: $scope.quiz.questions
```
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: it's just as it should be.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 4]
```diff
@@ -32,3 +32,28 @@ quizzesControllers.controller("indexQuizCtrl", ["$scope", "Quizzes",
...
 ]);
+
+quizzesControllers.controller('editQuizCtrl',
+  ['$scope', '$routeParams', '$location', 'Quiz',
+  function($scope, $routeParams, $location, Quiz) {
+    $scope.quiz = Quiz.show({id: $routeParams.quizId});
+
+    $scope.add = function(){
+      $scope.quiz.questions.push({
+        content: ''
```
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js [hunk 5]
```diff
@@ -7,3 +7,12 @@ videoQuizServices.factory("Quizzes", ["$resource",
     });
   }
 ]);
+
+videoQuizServices.factory('Quiz', ['$resource',
+  function($resource){
+    return $resource('quizzes/:id.json', {}, {
+      show: {method: 'GET', params: {id: '@id'}},
```
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js [hunk 6]
```diff
@@ -7,3 +7,12 @@ videoQuizServices.factory("Quizzes", ["$resource",
     });
   }
 ]);
+
+videoQuizServices.factory('Quiz', ['$resource',
```
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js [hunk 7]
```diff
@@ -7,3 +7,12 @@ videoQuizServices.factory("Quizzes", ["$resource",
...
+  function($resource){
+    return $resource('quizzes/:id.json', {}, {
```
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 8]
```diff
@@ -32,3 +32,28 @@ quizzesControllers.controller("indexQuizCtrl", ["$scope", "Quizzes",
...
+
+    $scope.editQuiz = function() {
+      Quiz.update({
+        id: $routeParams.quizId,
+        quiz: {
+          name: $scope.quiz.name,
+          questions_attributes: $scope.quiz.questions
+        }
+      }, function() {
+        $location.path('/quizzes');
```
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 9]
```diff
@@ -99,3 +99,35 @@ describe("indexQuizCtrl", function(){
     });
   });
 });
+
+describe('editQuizCtrl', function() {
```
Comment (OKrain) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js [hunk 10]
```diff
@@ -7,3 +7,12 @@ videoQuizServices.factory("Quizzes", ["$resource",
...
+      show: {method: 'GET', params: {id: '@id'}},
+      update: {method:'PUT', params: {id: '@id'}}
```
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 11]
```diff
@@ -99,3 +99,35 @@ describe("indexQuizCtrl", function(){
...
+  var $controller, $scope, $routeParams, Quiz;
+
+  $scope = {};
+  $routeParams = {};
+  Quiz = {
+    show: function() {},
+    update: function() {}
+  };
+
+  beforeEach(module('quizzesControllers'));
```
Comment (OKrain) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 12]
```diff
@@ -99,3 +99,35 @@ describe("indexQuizCtrl", function(){
...
+  beforeEach(inject(function(_$controller_){
+    $controller = _$controller_;
+
+    $controller('editQuizCtrl', {
```
Comment (OKrain) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 13]
```diff
@@ -99,3 +99,35 @@ describe("indexQuizCtrl", function(){
...
+      $scope: $scope,
+      $routeParams: $routeParams,
+      Quiz: Quiz
+    });
+
+    $scope.quiz = {};
+  }));
+
+  describe('$scope.editQuiz', function() {
```
Comment (OKrain) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 14]
```diff
@@ -99,3 +99,35 @@ describe("indexQuizCtrl", function(){
...
+    it('Update new input to the form', function() {
```
Comment (OKrain) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: should i replace single quotes with double quotes? single ones seem to be more appropriate.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 15]
```diff
@@ -99,3 +99,35 @@ describe("indexQuizCtrl", function(){
...
+      spyOn(Quiz, 'update');
```
Comment (OKrain) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: it's just as it should be.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 16]
```diff
@@ -32,3 +32,28 @@ quizzesControllers.controller("indexQuizCtrl", ["$scope", "Quizzes",
...
+      });
+    };
+  }]
+);
```
Comment (superNazo) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: add a free line to the end

**Comment by superNazo** [comment]
everything is fine, except:
add free line at the end of quizzesController.js (59)
and I think yes it would be better to exchange all the quotes with one type (single or double)

**Comment by anatoliiibrahimov** [comment]
take into account all comments of bot , everything else is fine

**Comment by OKrain** [comment]
> add free line at the end of quizzesController.js (59)
Done

> and I think yes it would be better to exchange all the quotes with one type (single or double)
But how? Single quote or double quote?

**Comment by vovka** [comment]
:ok: for demo. Do not resolve task in jira.

**Comment by OKrain** [comment]
> :ok: for demo. Do not resolve task in jira.

something wrong?

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 1 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Do not change quotes.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 17]
```diff
@@ -99,3 +99,35 @@ describe("indexQuizCtrl", function(){
...
+  $scope = {};
+  $routeParams = {};
+  Quiz = {
+    show: function() {},
+    update: function() {}
+  };
+
+  beforeEach(module('quizzesControllers'));
+  beforeEach(inject(function(_$controller_){
+    $controller = _$controller_;
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: You use `$controlle` variable only in this current scope. So you may use `_$controller_ ` directly from arguments list. No need to assign another variable in this particular case.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 18]
```diff
@@ -99,3 +99,35 @@ describe("indexQuizCtrl", function(){
...
+
+    $controller('editQuizCtrl', {
+      $scope: $scope,
+      $routeParams: $routeParams,
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: You use `$routeParams` only in this line. So you may just pass `{}` here without assigning a separate variable.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 14 above)
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Please, split test phases.

File: app/assets/javascripts/angularjs_app/components/quizzes/views/indexQuiz.html [hunk 19]
```diff
@@ -25,7 +25,7 @@
       </div>
       <div class="col-xs-5">
         <a href="#" class="link-to btn btn-success">Show</a>
-        <a href="#" class="link-to btn btn-info">Edit</a>
+        <a href="quizzes#/{{quiz.id}}/edit_quiz" class="link-to btn btn-info">Edit</a>
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/views/indexQuiz.html]: Could you make this route more restful? Like `/quizzes/:id/edit`?

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js (hunk 5 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js]: Is it possible to replace these custom methods with standard ones - `get` and `save`?

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 14 above)
Comment (OKrain) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Done.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js (hunk 5 above)
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesServices.js]: No.
In this case we should use custom methods.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 1 above)
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: ok

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 17 above)
Comment (OKrain) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Done.

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 18 above)
Comment (OKrain) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Done.

File: app/assets/javascripts/angularjs_app/components/quizzes/views/indexQuiz.html (hunk 19 above)
Comment (OKrain) [inline app/assets/javascripts/angularjs_app/components/quizzes/views/indexQuiz.html]: Yeah its make sense.
Done.

**Pull request merged**